corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -27,37 +28,73 @@ hqDefine("data_dictionary/js/data_dictionary", [
         self.name = name || gettext("No Name");
         self.url = "#" + name;
         self.fhirResourceType = ko.observable(fhirResourceType);
-        self.properties = ko.observableArray();
+        self.groups = ko.observableArray();
+
+        self.init = function (groupData, changeSaveButton) {
+            for (let group of groupData) {
+                let groupObj = groupsViewModel(group.id, group.name, group.description, group.index, self.name, group.deprecated);
+                if (group.id) {
+                    groupObj.name.subscribe(changeSaveButton);
+                    groupObj.description.subscribe(changeSaveButton);
+                    groupObj.deprecated.subscribe(changeSaveButton);
+                }
+                groupObj.properties.subscribe(changeSaveButton);
 
-        self.init = function (groupDict, changeSaveButton) {
-            _.each(groupDict, function (properties, group) {
-                var groupObj = propertyListItem(group, '', true, group, self.name);
-                self.properties.push(groupObj);
-                properties.sort((a, b) => a.index - b.index);
-                _.each(properties, function (prop) {
+                for (let prop of group.properties) {
                     var propObj = propertyListItem(prop.name, prop.label, false, prop.group, self.name, prop.data_type,
                         prop.description, prop.allowed_values, prop.fhir_resource_prop_path, prop.deprecated,
-                        prop.removeFHIRResourcePropertyPath);
+                        prop.removeFHIRResourcePropertyPath, prop.index);
                     propObj.description.subscribe(changeSaveButton);
                     propObj.label.subscribe(changeSaveButton);
                     propObj.fhirResourcePropPath.subscribe(changeSaveButton);
                     propObj.dataType.subscribe(changeSaveButton);
                     propObj.deprecated.subscribe(changeSaveButton);
                     propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
                     propObj.allowedValues.on('change', changeSaveButton);
-                    self.properties.push(propObj);
-                });
+                    groupObj.properties.push(propObj);
+                }
+
+                groupObj.properties.sort((a, b) => a.index - b.index);
```

Comment:
aren't they already sorted out when they are fetched from DB?

Reply:
Yeah, this won't be needed anymore. I have removed this thanks for pointing this out. [14118d9](https://github.com/dimagi/commcare-hq/pull/32878/commits/14118d9b929e6e3a13526a65d933d2f1a0cb810c)